src/Orleans/AssemblyLoader/AssemblyCatalog.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Orleans.Runtime
+{
+    public class AssemblyCatalog : IAssemblyCatalog
```

Comment:
@galvesribeiro How long can this branch wait? :-P One the first though this modification would seem sensible, but would require one to suss out the details with APIs. I'm glad if you consider this and find a way, or someone else, but as of myself, well, you'd need to wait a week or two at minimum. Though if this makes sense, I, you, or someone else, could do a PR later.